graylog2-web-interface/src/components/common/SearchForm.jsx

```diff
@@ -88,27 +103,37 @@ class SearchForm extends React.Component {
   _onSearch = (e) => {
     e.preventDefault();
 
-    this._setLoadingState();
-    this.props.onSearch(this.refs.query.value, this._resetLoadingState);
+    this._setLoadingState().then(() => {
+      this.props.onSearch(this.state.query, this._resetLoadingState);
```

Comment:
I would add a little comment here or in `_setLoadingState` to clarify why we use a promise in here.

Reply:
Good point!